### ISSUE TITLE
Suggestion: automatically dropback to avoid error

### ISSUE THREAD

**Issue by pikakolendo02** [jmcarpenter2/swifter#212]
for some functions, when using `swifter.apply()`,
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/mambaforge/lib/python3.11/site-packages/swifter/swifter.py:241, in SeriesAccessor._dask_apply(self, func, convert_dtype, *args, **kwds)
    237 with suppress_stdout_stderr_logging():
    238     tmp_df = (
    239         dd.from_pandas(sample, npartitions=self._npartitions)
    240         .map_partitions(func, *args, meta=meta, **kwds)
--> 241         .compute(scheduler=self._scheduler)
    242     )
    243     self._validate_apply(
    244         tmp_df.equals(meta),
    245         error_message=("Dask map-partitions sample does not match pandas apply sample."),
    246     )

File ~/mambaforge/lib/python3.11/site-packages/dask/base.py:310, in DaskMethodsMixin.compute(self, **kwargs)
    287 """Compute this dask collection
    288 
    289 This turns a lazy Dask collection into its in-memory equivalent.
   (...)
    308 dask.compute
    309 """
--> 310 (result,) = compute(self, traverse=False, **kwargs)
    311 return result

File ~/mambaforge/lib/python3.11/site-packages/dask/base.py:595, in compute(traverse, optimize_graph, scheduler, get, *args, **kwargs)
    593     postcomputes.append(x.__dask_postcompute__())
--> 595 results = schedule(dsk, keys, **kwargs)
    596 return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])

File ~/mambaforge/lib/python3.11/site-packages/dask/multiprocessing.py:233, in get(dsk, keys, num_workers, func_loads, func_dumps, optimize_graph, pool, initializer, chunksize, **kwargs)
    231 try:
    232     # Run
--> 233     result = get_async(
    234         pool.submit,
    235         pool._max_workers,
    236         dsk3,
    237         keys,
    238         get_id=_process_get_id,
    239         dumps=dumps,
    240         loads=loads,
    241         pack_exception=pack_exception,
    242         raise_exception=reraise,
    243         chunksize=chunksize,
    244         **kwargs,
    245     )
    246 finally:

File ~/mambaforge/lib/python3.11/site-packages/dask/local.py:499, in get_async(submit, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, chunksize, **kwargs)
    498 while state["waiting"] or state["ready"] or state["running"]:
--> 499     fire_tasks(chunksize)
    500     for key, res_info, failed in queue_get(queue).result():

File ~/mambaforge/lib/python3.11/site-packages/dask/local.py:481, in get_async.<locals>.fire_tasks(chunksize)
    475     data = {
    476         dep: state["cache"][dep] for dep in get_dependencies(dsk, key)
    477     }
    478     args.append(
    479         (
    480             key,
--> 481             dumps((dsk[key], data)),
    482             dumps,
    483             loads,
    484             get_id,
    485             pack_exception,
    486         )
    487     )
    489 # Batch submit

File ~/mambaforge/lib/python3.11/site-packages/cloudpickle/cloudpickle_fast.py:73, in dumps(obj, protocol, buffer_callback)
     70 cp = CloudPickler(
     71     file, protocol=protocol, buffer_callback=buffer_callback
     72 )
---> 73 cp.dump(obj)
     74 return file.getvalue()

File ~/mambaforge/lib/python3.11/site-packages/cloudpickle/cloudpickle_fast.py:632, in CloudPickler.dump(self, obj)
    631 try:
--> 632     return Pickler.dump(self, obj)
    633 except RuntimeError as e:

TypeError: cannot pickle 'Boost.Python.function' object

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
File <timed exec>:25

File ~/mambaforge/lib/python3.11/site-packages/swifter/swifter.py:296, in SeriesAccessor.apply(self, func, convert_dtype, args, **kwds)
    294 # If parallel processing is forced by the user, then skip the logic and apply dask
    295 if self._force_parallel:
--> 296     return self._dask_apply(func, convert_dtype, *args, **kwds)
    298 sample = self._obj.iloc[self._SAMPLE_INDEX]
    299 # check if input is string or
    300 # if the user is overriding the string processing default

File ~/mambaforge/lib/python3.11/site-packages/swifter/swifter.py:272, in SeriesAccessor._dask_apply(self, func, convert_dtype, *args, **kwds)
    263 if self._progress_bar:
    264     with TQDMDaskProgressBar(desc=self._progress_bar_desc or "Dask Apply"):
    265         return (
    266             dd.from_pandas(self._obj, npartitions=self._npartitions)
    267             .apply(
    268                 lambda x: func(x, *args, **kwds),
    269                 convert_dtype=convert_dtype,
    270                 meta=meta,
    271             )
--> 272             .compute(scheduler=self._scheduler)
    273         )
    274 else:
    275     return (
    276         dd.from_pandas(self._obj, npartitions=self._npartitions)
    277         .apply(
   (...)
    282         .compute(scheduler=self._scheduler)
    283     )

File ~/mambaforge/lib/python3.11/site-packages/dask/base.py:310, in DaskMethodsMixin.compute(self, **kwargs)
    286 def compute(self, **kwargs):
    287     """Compute this dask collection
    288 
    289     This turns a lazy Dask collection into its in-memory equivalent.
   (...)
    308     dask.compute
    309     """
--> 310     (result,) = compute(self, traverse=False, **kwargs)
    311     return result

File ~/mambaforge/lib/python3.11/site-packages/dask/base.py:595, in compute(traverse, optimize_graph, scheduler, get, *args, **kwargs)
    592     keys.append(x.__dask_keys__())
    593     postcomputes.append(x.__dask_postcompute__())
--> 595 results = schedule(dsk, keys, **kwargs)
    596 return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])

File ~/mambaforge/lib/python3.11/site-packages/dask/multiprocessing.py:233, in get(dsk, keys, num_workers, func_loads, func_dumps, optimize_graph, pool, initializer, chunksize, **kwargs)
    228 # Note former versions used a multiprocessing Manager to share
    229 # a Queue between parent and workers, but this is fragile on Windows
    230 # (issue #1652).
    231 try:
    232     # Run
--> 233     result = get_async(
    234         pool.submit,
    235         pool._max_workers,
    236         dsk3,
    237         keys,
    238         get_id=_process_get_id,
    239         dumps=dumps,
    240         loads=loads,
    241         pack_exception=pack_exception,
    242         raise_exception=reraise,
    243         chunksize=chunksize,
    244         **kwargs,
    245     )
    246 finally:
    247     if cleanup:

File ~/mambaforge/lib/python3.11/site-packages/dask/local.py:499, in get_async(submit, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, chunksize, **kwargs)
    497 # Main loop, wait on tasks to finish, insert new ones
    498 while state["waiting"] or state["ready"] or state["running"]:
--> 499     fire_tasks(chunksize)
    500     for key, res_info, failed in queue_get(queue).result():
    501         if failed:

File ~/mambaforge/lib/python3.11/site-packages/dask/local.py:481, in get_async.<locals>.fire_tasks(chunksize)
    474     # Prep args to send
    475     data = {
    476         dep: state["cache"][dep] for dep in get_dependencies(dsk, key)
    477     }
    478     args.append(
    479         (
    480             key,
--> 481             dumps((dsk[key], data)),
    482             dumps,
    483             loads,
    484             get_id,
    485             pack_exception,
    486         )
    487     )
    489 # Batch submit
    490 for i in range(-(len(args) // -chunksize)):

File ~/mambaforge/lib/python3.11/site-packages/cloudpickle/cloudpickle_fast.py:73, in dumps(obj, protocol, buffer_callback)
     69 with io.BytesIO() as file:
     70     cp = CloudPickler(
     71         file, protocol=protocol, buffer_callback=buffer_callback
     72     )
---> 73     cp.dump(obj)
     74     return file.getvalue()

File ~/mambaforge/lib/python3.11/site-packages/cloudpickle/cloudpickle_fast.py:632, in CloudPickler.dump(self, obj)
    630 def dump(self, obj):
    631     try:
--> 632         return Pickler.dump(self, obj)
    633     except RuntimeError as e:
    634         if "recursion" in e.args[0]:

TypeError: cannot pickle 'Boost.Python.function' object
```
I think it is better to dropback to another parallel method or single-process, rather than raise an error.

**Comment by jmcarpenter2** [comment]
ah good call. interesting enough it does already fallback from `ddf.map_partitions` -> `ddf.apply` (still parallel, but slower).. i think there's one additional fallback that could happen from `ddf.apply` -> `df.apply` (single-threaded)

**Comment by jmcarpenter2** [comment]
Resolved with Release 1.3.5
